cli/src/main/java/com/walmartlabs/concord/cli/runner/secrets/RemoteSecretsProvider.java

```diff
@@ -0,0 +1,165 @@
+package com.walmartlabs.concord.cli.runner.secrets;
+
+/*-
+ * *****
+ * Concord
+ * -----
+ * Copyright (C) 2017 - 2025 Walmart Inc.
+ * -----
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =====
+ */
+
+import com.walmartlabs.concord.client2.*;
+import com.walmartlabs.concord.common.secret.BinaryDataSecret;
+import com.walmartlabs.concord.common.secret.KeyPair;
+import com.walmartlabs.concord.common.secret.UsernamePassword;
+import com.walmartlabs.concord.runtime.v2.sdk.SecretService;
+
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Objects;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class RemoteSecretsProvider implements SecretsProvider {
+
+    private final Path workDir;
+    private final SecretClient secretClient;
+
+    public RemoteSecretsProvider(Path workDir, String baseUrl, String apiKey) {
+        this.workDir = requireNonNull(workDir);
+
+        var apiClient = new DefaultApiClientFactory(requireNonNull(baseUrl))
+                .create(ApiClientConfiguration.builder()
```

Comment:
I think it may be useful to set a user agent on the client here to identify calls from the CLI in a real environment.

```java
        var apiClient = new DefaultApiClientFactory(requireNonNull(baseUrl))
                .create(ApiClientConfiguration.builder()
                        .apiKey(requireNonNull(apiKey))
                        .build())
                .setUserAgent("Concord-CLI");
```

Reply:
Good idea, will do.